### ISSUE TITLE
Can't specify order of execution in the transaction API

### ISSUE THREAD

**Issue by joaopaulobdac** [prisma/prisma#4563]
## Problem

When running a transaction, the order of execution is not the same as the order of the items in the array parameter.

In this example, `db.thing.findMany` is being executed before `$executeRaw`. It doesn't matter if I change the positions in the array.

```ts
await db.$transaction([
    db.$executeRaw(`SET LOCAL api.userId = ${userId}`),
    db.thing.findMany()
])
```

## Context

I'm trying to use postgresql's row security policies. For more context, please read https://github.com/prisma/prisma/issues/1844#issuecomment-737024796

I have a postgres policy that will restrict the prisma client based on the `userId` coming from the api request:

```sql
\c devdb;

CREATE USER api WITH PASSWORD 'api';



CREATE TABLE "User" (
  "id" TEXT PRIMARY KEY NOT NULL
);

GRANT SELECT ON "User" TO api;



CREATE TABLE "Thing" (
  "id" SERIAL PRIMARY KEY NOT NULL,
  "name" TEXT NOT NULL,
  "createdByUserId" TEXT NOT NULL REFERENCES "User"("id")
);

GRANT SELECT ON "Thing" TO api;

-- Will only let the prisma client select rows where the
-- id from the api request matches createdByUserId

ALTER TABLE "Thing" ENABLE ROW LEVEL SECURITY;

CREATE POLICY "ThingPolicy" ON "Thing" TO api USING (
    "Thing"."createdByUserId" = current_setting('api.userId')
);



INSERT INTO "User"("id") VALUES
  ('user1'),
  ('user2');

INSERT INTO "Thing"("name", "createdByUserId") VALUES
  ('Thing belonging to user1', 'user1'),
  ('Thing belonging to user2', 'user2');
```

As we can see from the schema above, postgres expects the `api.userId` to be set in order to query the table.

What I'm expecting to be able to do here is use the transaction api to achieve this transaction:

```sql
BEGIN;
SET LOCAL api.userId = ${userId}; -- takes effect for only the current transaction
${query};
COMMIT;
```

But it seems that these two client calls are not being run in a transaction in order. They are simply being run in parallel using `Promise.all`. Notice how there's no `BEGIN` in the logs:

```
prisma:info  Starting a postgresql pool with 1 connections.
prisma:info  Started http server on http+unix:///tmp/prisma-e47982a844395648af6cbc3e.sock
prisma:query SELECT "public"."Thing"."id", "public"."Thing"."name", "public"."Thing"."createdByUserId" FROM "public"."Thing" WHERE 1=1 OFFSET $1
prisma:query SET LOCAL api.userId = user1
PrismaClientUnknownRequestError2: 
Invalid `prisma.thing.findMany()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42704"), message: "unrecognized configuration parameter \"api.userId\"", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("guc.c"), line: Some(8965), routine: Some("GetConfigOptionByName") }) }) })
    at PrismaClientFetcher.request (/home/jp/.vscode/projects/temp/row-policies/node_modules/@prisma/client/runtime/index.js:79359:15)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async Promise.all (index 1)
    at Server.<anonymous> (/home/jp/.vscode/projects/temp/row-policies/src/main.ts:78:25) {
  clientVersion: '2.12.1'
}
```

## Suggested solution

Let users tell the transaction api that the calls passed in must be run in order and truly inside a transaction. Or maybe make that the default.

## Alternatives

I can solve this by creating a new connection for each request and setting the `userId` parameter for the connection, instead of for only a transaction:

```ts
const db = new PrismaClient()

await db.$executeRaw(`SET api.userId = ${userId}`)

const result = await db.thing.findMany()

await db.$disconnect()
```

But as I showed in https://github.com/prisma/prisma/issues/1844#issuecomment-737024796 this is very bad for performance.

**Comment by pantharshit00** [comment]
This is looking like a duplicate of https://github.com/prisma/prisma/issues/4269 and this bug report https://github.com/prisma/prisma/issues/4331

Feel free to comment if you feel those two doesn't cover this case.

**Comment by joaopaulobdac** [comment]
I should've search before opening this issue. My bad!
